geonode_mapstore_client/client/js/epics/gnsave.js

```diff
@@ -128,22 +129,33 @@ const SaveAPI = {
                 }
                 return response.data;
             });
+    },
+    document: (state, id, metadata) => {
+        const body = {
+            'title': metadata.name,
+            'abstract': metadata.description,
+            'thumbnail_url': metadata.thumbnail
+        };
+        return id ? updateDocument(id, body) : false;
+
     }
 };
 
 export const gnSaveContent = (action$, store) =>
     action$.ofType(SAVE_CONTENT)
         .switchMap((action) => {
             const state = store.getState();
-            const contentType = state.gnresource?.type || 'map';
+            const contentType = state.gnresource?.type || state.gnresource?.data.resource_type || 'map';
```

Comment:
`state.gnresource?.type` is a property we set client side once the viewer open (eg: https://github.com/GeoNode/geonode-mapstore-client/blob/6b874a2305cc86735dfbc6e291c0490ff19055b4/geonode_mapstore_client/client/js/epics/gnviewer.js#L149).
 I think we should do the same for document config here
https://github.com/GeoNode/geonode-mapstore-client/blob/6b874a2305cc86735dfbc6e291c0490ff19055b4/geonode_mapstore_client/client/js/epics/gnviewer.js#L187-L202
 and do not rely on the resource_type from api

Reply:
ok, I implement the setResourceType